src/main.cc

```diff
@@ -209,21 +209,20 @@ int main(int argc, char* argv[]) {
 #else
   for (const auto& filename: filenames_in) {
     filename_in = filename;
-    bool has_error = false;
     pid_t pid = fork();
     if (pid < 0) {
       Error("fork error");
     } else if (pid == 0) {
       // Do work in child process
       return RunWgtcc();
-    } else {
-      int stat;
-      wait(&stat);
-      has_error = has_error || (WIFEXITED(stat) && WEXITSTATUS(stat) != 0);
     }
+  }
 
-    if (has_error)
-      return 0;
+  for (size_t i = 0; i < filenames_in.size(); ++i) {
+      int stat; 
+      wait(&stat);
+      if (WIFEXITED(stat) && WEXITSTATUS(stat) != 0)
```

Comment:
Here is shit code that we both didn't notice.
Why not `WIFEXITED(stat) && WEXITSTATUS(stat)` or `WIFEXITED(stat) != 0 && WEXITSTATUS(stat) != 0` ?

Reply:
And, it should be `||` rather than `&&`, right?

Reply:
The right version should be `!WIFEXITED(stat) && WEXITSTATUS(stat)`, right ?

Reply:
http://www.gnu.org/software/libc/manual/html_node/Process-Completion-Status.html